frontend/src/components/Sidebar/SidebarAddClassModal.tsx

```diff
@@ -0,0 +1,257 @@
+import React from "react";
+import styled from "styled-components";
+import {
+  ScheduleCourse,
+  DNDScheduleTerm,
+  DNDSchedule,
+  IRequiredCourse,
+} from "../../models/types";
+import { XButton } from "../common";
+import { Modal } from "@material-ui/core";
+import { connect } from "react-redux";
+import { Dispatch } from "redux";
+import { addClassesAction } from "../../state/actions/scheduleActions";
+import { getScheduleFromState } from "../../state";
+import { AppState } from "../../state/reducers/state";
+import { fetchCourse } from "../../api";
+
+const InnerSection = styled.section`
+  position: fixed;
+  background: white;
+  width: 30%;
+  height: auto;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  padding: 8px;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const CloseButtonWrapper = styled.div`
+  position: absolute;
+  top: 12px;
+  right: 18px;
+`;
+
+const StyledLabel = styled.label`
+  display: flex;
+  flex-direction: column;
+`;
+
+const FormRow = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-around;
+`;
+
+const SubmitButton = styled.button`
+  margin: 8px;
+`;
+
+interface ReduxStoreSidebarAddClassModalProps {
+  schedule: DNDSchedule;
+}
+
+interface ReduxDispatchSidebarAddClassModalProps {
+  handleAddClasses: (
+    courses: ScheduleCourse[],
+    semester: DNDScheduleTerm
+  ) => void;
+}
+
+interface SidebarAddClassModalProps {
+  handleClose: () => void;
+  handleSubmit: (courses: ScheduleCourse[]) => void;
+  visible: boolean;
+  selectedCourses: IRequiredCourse[];
+}
+
+type Props = SidebarAddClassModalProps &
+  ReduxStoreSidebarAddClassModalProps &
+  ReduxDispatchSidebarAddClassModalProps;
+
+interface SidebarAddClassModalState {
+  queuedCourses: ScheduleCourse[];
+  formSemester: DNDScheduleTerm;
+  semesters: DNDScheduleTerm[];
+  courses: ScheduleCourse[];
+}
+
+export class SidebarAddClassModalComponent extends React.Component<
+  Props,
+  SidebarAddClassModalState
+> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      queuedCourses: [],
+      formSemester: {
+        season: "FL",
+        year: 0,
+        termId: 0,
+        id: 0,
+        status: "INACTIVE",
+        classes: [],
+      },
+      semesters: [],
+      courses: [],
+    };
+  }
+
+  componentDidMount() {
+    this.mapScheduleToSemesters();
+  }
+
+  handleSemesterChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    this.setState({
+      formSemester: this.state.semesters[Number(event.target.value)],
+    });
+  }
+
+  async handleSubmit() {
+    await this.mapRequirementToSchedule();
+
+    this.props.handleAddClasses(
+      this.state.queuedCourses,
+      this.state.formSemester
+    );
+    this.prepareToClose();
+  }
+
+  prepareToClose() {
+    this.setState({
+      queuedCourses: [],
+    });
+    this.props.handleClose();
+  }
+
+  mapScheduleToSemesters() {
+    let semesterList = [];
```

Comment:
These variables should all be type annotated. Type annotations prevent many runtime bugs, by catching them at compile time and it also makes the code more comprehensible.